### PR TITLE
add enable sequential cpu offloading to other stable diffusion pipelines

### PR DESCRIPTION
**Pull request by piEsposito**
Follow up of #1024

Adds `enable_sequential_cpu_offload` method to `StableDiffusionInpaintPipeline` and `StableDiffusionImg2ImgPipeline` alongs with proper tests. 

Also renames the test on `StableDiffusionPipeline.enable_sequential_cpu_offload` to `test_stable_diffusion_pipeline_with_sequential_cpu_offloading` for clarity.

@patrickvonplaten here I'm just adding the `enable_sequential_cpu_offload` method to the img2img and inpainting pipelines for  Stable Diffusion.

### CODE REVIEW

**Comment by piEsposito** [comment]
@patrickvonplaten the tests that are not passing seem to be related to the `runwayml/stable-diffusion-v1-5`, which you can see I did not edit at all.

Same happened with https://github.com/huggingface/diffusers/pull/1010 btw.

**Comment by piEsposito** [comment]
@patrickvonplaten I've added @pedrogengo to [d6fcb91](https://github.com/huggingface/diffusers/pull/1085/commits/d6fcb917d17f40a72f0832dd96305fe2d78c3bae) because he found the bug that this commit solves.

File: tests/pipelines/stable_diffusion/test_stable_diffusion_img2img.py [hunk 1]
```diff
@@ -603,3 +603,47 @@ def test_callback_fn(step: int, timestep: int, latents: torch.FloatTensor) -> No
...
+        prompt = "A fantasy landscape, trending on artstation"
+
+        generator = torch.Generator(device=torch_device).manual_seed(0)
+        _ = pipe(
+            prompt=prompt,
+            init_image=init_image,
+            strength=0.75,
+            guidance_scale=7.5,
+            generator=generator,
+            output_type="np",
```
Comment (patrickvonplaten) [inline tests/pipelines/stable_diffusion/test_stable_diffusion_img2img.py:644]:
```suggestion
            output_type="np",
            num_inference_steps=5,
```

File: tests/pipelines/stable_diffusion/test_stable_diffusion_inpaint.py [hunk 2]
```diff
@@ -378,4 +378,48 @@ def test_stable_diffusion_inpaint_pipeline_pndm(self):
...
+        pipe.enable_sequential_cpu_offload()
+
+        prompt = "Face of a yellow cat, high resolution, sitting on a park bench"
+
+        generator = torch.Generator(device=torch_device).manual_seed(0)
+        _ = pipe(
+            prompt=prompt,
+            image=init_image,
+            mask_image=mask_image,
+            generator=generator,
```
Comment (patrickvonplaten) [inline tests/pipelines/stable_diffusion/test_stable_diffusion_inpaint.py:419]:
```suggestion
            generator=generator,
            num_inference_steps=5,
```

**Review by patrickvonplaten — APPROVED** [review]

**Comment by piEsposito** [comment]
@patrickvonplaten thank you for the kindness of addreessing your review.

**Comment by patrickvonplaten** [comment]
Thanks a lot the PR :-)

**Pull request merged**